woocommerce/woocommerce-ios
# [Mobile Payments] Measure Order Creation IPP Flows

## Description

We track here the duration of Order Creation and IPP flows to gather good baseline numbers to evaluate the improvements of the Speedy IPP projects.

We measure:

Flow 1: Add new order button tapped to order was successfully created.
Flow 2: Add new order button tapped to card payment method was selected.
Flow 3: Add new order button tapped to card payment collection was successful (in both regular and interac).
Flow 4: Card payment method was selected to Card payment collection was successful (in both regular and interac).

We reported those values by adding a new properties to the related events:

| Event      | New properties |
| ----------- | ----------- |
| `"order_creation_success" `     | `"milliseconds_since_order_add_new"`       |
| `"payments_flow_collect"`   | `"milliseconds_since_order_add_new"`        |
|`"card_present_collect_payment_success"` |  `"milliseconds_since_order_add_new"`, `"milliseconds_since_card_collect_payment_flow"`|

## Testing instructions


1. Go to Orders
2. Tap on New
3. Add products to the order, and tap on Create
4. Tap on Collect payment
5. Follow the flow to collect the payment with card

Check that the events are tracked, e.g.:
`"order_creation_success" `:

`🔵 Tracked order_creation_success, properties: [AnyHashable("blog_id"): 214354650, AnyHashable("milliseconds_since_order_add_new"): 17492, AnyHashable("is_wpcom_store"): true]`

`"payments_flow_collect"`:

` 🔵 Tracked payments_flow_collect, properties: [AnyHashable("flow"): "order_payment", AnyHashable("milliseconds_since_order_add_new"): 22675, AnyHashable("payment_method"): "card", AnyHashable("blog_id"): 214354650, AnyHashable("is_wpcom_store"): true]`

`card_present_collect_payment_success`:

`🔵 Tracked card_present_collect_payment_success, properties: [AnyHashable("is_wpcom_store"): true, AnyHashable("country"): "US", AnyHashable("plugin_slug"): "woocommerce-payments", AnyHashable("milliseconds_since_order_add_new"): 83651, AnyHashable("milliseconds_since_card_collect_payment_flow"): 32618, AnyHashable("blog_id"): 214354650, AnyHashable("payment_method_type"): "card", AnyHashable("card_reader_model"): "COTS_DEVICE"]`

Repeat the process to check that the measurements are properly resetted after a successful order.

## Screenshots



---
